experiment/simulation/js/main.js

```diff
@@ -1 +1,579 @@
+  ctx.lineTo(maxx, mxy);
+  ctx.stroke();
+  ctx.beginPath();
+  ctx.lineWidth = 2;
+  ctx.strokeStyle = "red";
+  ctx.moveTo(mnx, maxy);
+  ctx.lineTo(mxx, maxy);
+  ctx.stroke();
+}
+function setParameters() {
+  // get  & set the coordinates from the input form
+  let x1 = Number(document.getElementById("x1").value);
+  let y1 = Number(document.getElementById("y1").value);
+  let x2 = Number(document.getElementById("x2").value);
+  let y2 = Number(document.getElementById("y2").value);
+
+  x1 = x1 + Number(originx);
+  y1 = y1 + Number(originy);
+  x2 = x2 + Number(originx);
+  y2 = y2 + Number(originy);
+
+  point1 = [x1, y1];
+  point2 = [x2, y2];
+  flag = 1;
+  dy = Math.abs(y2 - y1);
+  dx = Math.abs(x2 - x1);
+  currx = point1[0];
+  curry = point1[1];
+  if (point1[0] == point2[0]) {
+    if (point1[1] > point2[1]) slope = Number.MIN_SAFE_INTEGER;
+    else slope = Number.MAX_SAFE_INTEGER;
+  } else {
+    slope = (point2[1] - point1[1]) / (point2[0] - point1[0]);
+    if (slope > 1) {
+      // set the parameter
+      decision_parameter = 2 * dx - dy;
+    } else if (slope >= 0 && slope <= 1) {
+      decision_parameter = 2 * dy - dx;
+    } else if (slope >= -1 && slope < 0) {
+      decision_parameter = 2 * Math.abs(dy) - dx;
+    } else if (slope < -1) {
+      decision_parameter = 2 * Math.abs(dx) - Math.abs(dy);
+    }
+  }
+  // dp.push(decision_parameter);
+  return;
+}
+
+function highlight(x, y, color) {
+  let X = (x + 0.5) * block_size;
+  let Y = height - (y + 0.5) * block_size;
+  ctx.beginPath();
+  ctx.fillStyle = color;
+  ctx.fillRect(X, Y, height / (2 * divisions), height / (2 * divisions));
+  ctx.fillRect(X, Y, -(height / (2 * divisions)), -(height / (2 * divisions)));
+  ctx.fillRect(X, Y, -(height / (2 * divisions)), height / (2 * divisions));
+  ctx.fillRect(X, Y, height / (2 * divisions), -(height / (2 * divisions)));
+  ctx.stroke();
+}
+
+function drawGrid() {
+  // parameters set by now
+  // draw the grid
+  ctx.beginPath();
+  ctx.strokeStyle = "grey";
+  ctx.lineWidth = 0.5;
+  for (var i = 0; i <= width; i += height / divisions) {
+    ctx.moveTo(i, 0);
+    ctx.lineTo(i, height);
+  }
+
+  for (var i = 0; i <= width; i += height / divisions) {
+    ctx.moveTo(0, i);
+    ctx.lineTo(width, i);
+  }
+  ctx.stroke();
+
+  let X1 = (point1[0] + 0.5) * block_size;
+  let Y1 = height - (point1[1] + 0.5) * block_size;
+  let X2 = (point2[0] + 0.5) * block_size;
+  let Y2 = height - (point2[1] + 0.5) * block_size;
+  ctx.beginPath();
+  ctx.arc(X1, Y1, 1, 0, 2 * Math.PI, false);
+  ctx.lineWidth = 3;
+  ctx.strokeStyle = "red";
+  ctx.stroke();
+  ctx.beginPath();
+  ctx.arc(X2, Y2, 1, 0, 2 * Math.PI, false);
+  ctx.strokeStyle = "red";
+  ctx.lineWidth = 5;
+  ctx.stroke();
+  makeAxis();
+
+  ctx.beginPath();
+  ctx.lineWidth = 2;
+  ctx.strokeStyle = "yellow";
+  ctx.moveTo(X1, Y1);
+  ctx.lineTo(X2, Y2);
+  ctx.stroke();
+}
+
+function handleNext() {
+  dp.push(decision_parameter);
```

Comment:
Please try to minimize the use of global variables by explicitly passing variables into functions when necesary. Analysing the code becomes much mode difficult if global variables are split across multiple functions with no limits on scope.
Global variables should strictly be exceptions and not the norm